src/libraries/System.Collections.Concurrent/src/System/Collections/Concurrent/ConcurrentDictionary.cs

```diff
@@ -430,55 +398,126 @@ private bool TryRemoveInternal(TKey key, [MaybeNullWhen(false)] out TValue value
                     }
                 }
 
-                value = default(TValue)!;
+                value = default;
                 return false;
             }
         }
 
         /// <summary>
-        /// Attempts to get the value associated with the specified key from the <see
-        /// cref="ConcurrentDictionary{TKey,TValue}"/>.
+        /// Attempts to get the value associated with the specified key from the <see cref="ConcurrentDictionary{TKey,TValue}"/>.
         /// </summary>
         /// <param name="key">The key of the value to get.</param>
-        /// <param name="value">When this method returns, <paramref name="value"/> contains the object from
-        /// the
-        /// <see cref="ConcurrentDictionary{TKey,TValue}"/> with the specified key or the default value of
-        /// <typeparamref name="TValue"/>, if the operation failed.</param>
-        /// <returns>true if the key was found in the <see cref="ConcurrentDictionary{TKey,TValue}"/>;
-        /// otherwise, false.</returns>
-        /// <exception cref="System.ArgumentNullException"><paramref name="key"/> is a null reference
-        /// (Nothing in Visual Basic).</exception>
+        /// <param name="value">
+        /// When this method returns, <paramref name="value"/> contains the object from
+        /// the <see cref="ConcurrentDictionary{TKey,TValue}"/> with the specified key or the default value of
+        /// <typeparamref name="TValue"/>, if the operation failed.
+        /// </param>
+        /// <returns>true if the key was found in the <see cref="ConcurrentDictionary{TKey,TValue}"/>; otherwise, false.</returns>
+        /// <exception cref="ArgumentNullException"><paramref name="key"/> is a null reference (Nothing in Visual Basic).</exception>
         public bool TryGetValue(TKey key, [MaybeNullWhen(false)] out TValue value)
         {
-            if (key == null) ThrowKeyNullException();
-            return TryGetValueInternal(key, _comparer.GetHashCode(key), out value);
```

Comment:
@stephentoub Why did you choose to duplicate the logic that is in `TryGetValueInternal(...)`? Is this worth a comment?

Reply:
Do you mean the duplication between TryGetValue and TryGetValueInternal?  The difference is that TryGetValueInternal is used when we already have a hashcode, whereas TryGetValue is used when we don't and need to compute one.  In order to compute one, I'd need to go through the same decision tree that TryGetValueInternal does, which means we'd be doing it twice.  Rather than paying to do it twice, I duplicated the code, as TryGetValue is in most cases the hot path / most important to be fast member of ConcurrentDictionary.